[BUG] Still receiving warning: narrowing conversion from 'int' to 'short int'

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

The error happens in both **JyersUI** and **ProUI**

```r
Marlin/src/lcd/e3v2/jyersui/dwin.cpp: In member function 'void<unnamed struct>::drawBedMesh(int16_t, uint8_t, uint16_t, uint16_t)':
Marlin/src/lcd/e3v2/jyersui/dwin.cpp:392:51: warning: narrowing conversion of '(((int)(((short unsigned int)((int)cell_height_px)) / 2)) - 6)' from 'int' to 'short int' [-Wnarrowing]
  392 |           xy_int_t offset { 0, cell_height_px / 2 - 6 };
```
^JyersUI^
```r
Marlin\src\lcd\e3v2\proui\bedlevel_tools.cpp: In static member function 'static void BedLevelToolsClass::Draw_Bed_Mesh(int16_t, uint8_t, uint16_t, uint16_t)':
Marlin\src\lcd\e3v2\proui\bedlevel_tools.cpp:232:47: warning: narrowing conversion of '(((int)(((short unsigned int)((int)cell_height_px)) / 2)) - ((int)fs))' from 'int' to 'short int' [-Wnarrowing]
  232 |       xy_int_t offset { 0, cell_height_px / 2 - fs };
```
^ProUI^

I know @thinkyhead just updated the code, but the warning still happens.
before, the code `xy_int_t offset { 0, cell_height_px / 2 - fs };` was `xy_int8_t` and that also gave the warning.

I suggested it can be changed to `xy_long_t` because that gave no warning.

### Bug Timeline

started with 376673d - 🚸 Minor touch calibration improvement (#26445)

### Expected behavior

no warning is expected

### Actual behavior

warning is given

### Steps to Reproduce

1. Load Ender-3 V2 Configuration files
2. ANY(`AUTO_BED_LEVELING_BILINEAR`, `AUTO_BED_LEVELING_UBL`, `MESH_BED_LEVELING`)
3. `DWIN_LCD_PROUI` or `DWIN_CREALIT_LCD_JYERSUI`
4. compile

### Version of Marlin Firmware

bugfix-2.1.x

### Printer model

Ender-3V2, Aquila

### Electronics

_No response_

### LCD/Controller

4.3" DWIN

### Other add-ons

_No response_

### Bed Leveling

UBL Bilinear mesh

### Your Slicer

None

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[configurations files](https://github.com/MarlinFirmware/Configurations/tree/bugfix-2.1.x/config/examples/Creality/Ender-3%20V2/CrealityV422/CrealityUI)

--- a/Marlin/src/lcd/e3v2/jyersui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/jyersui/dwin.cpp
@@ -389,7 +389,8 @@ private:
 
         // Draw value text on
         if (viewer_print_value) {
-          xy_int_t offset { 0, cell_height_px / 2 - 6 };
+          xy_int8_t offset;
+          offset.y = cell_height_px / 2 - 6;
           if (isnan(bedlevel.z_values[x][y])) {  // undefined
             dwinDrawString(false, font6x12, COLOR_WHITE, COLOR_BG_BLUE, start_x_px + cell_width_px / 2 - 5, start_y_px + offset.y, F("X"));
           }

--- a/Marlin/src/lcd/e3v2/jyersui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/jyersui/dwin.cpp
@@ -389,8 +389,7 @@ private:
 
         // Draw value text on
         if (viewer_print_value) {
-          xy_int8_t offset;
-          offset.y = cell_height_px / 2 - 6;
+          xy_uint_t offset { 0, cell_height_px / 2 - 6 };
           if (isnan(bedlevel.z_values[x][y])) {  // undefined
             dwinDrawString(false, font6x12, COLOR_WHITE, COLOR_BG_BLUE, start_x_px + cell_width_px / 2 - 5, start_y_px + offset.y, F("X"));
           }

--- a/Marlin/src/lcd/e3v2/proui/bedlevel_tools.cpp
+++ b/Marlin/src/lcd/e3v2/proui/bedlevel_tools.cpp
@@ -211,9 +211,9 @@ bool BedLevelTools::meshValidate() {
 
   void BedLevelTools::drawBedMesh(int16_t selected/*=-1*/, uint8_t gridline_width/*=1*/, uint16_t padding_x/*=8*/, uint16_t padding_y_top/*=(40 + 53 - 7)*/) {
     drawing_mesh = true;
-    const uint16_t total_width_px = DWIN_WIDTH - padding_x - padding_x;
-    const uint16_t cell_width_px  = total_width_px / (GRID_MAX_POINTS_X);
-    const uint16_t cell_height_px = total_width_px / (GRID_MAX_POINTS_Y);
+    const uint16_t total_width_px = DWIN_WIDTH - padding_x - padding_x,
+                   cell_width_px  = total_width_px / (GRID_MAX_POINTS_X),
+                   cell_height_px = total_width_px / (GRID_MAX_POINTS_Y);
     const float v_max = abs(getMaxValue()), v_min = abs(getMinValue()), rmax = _MAX(v_min, v_max);
 
     // Clear background from previous selection and select new square
@@ -247,8 +247,7 @@ bool BedLevelTools::meshValidate() {
       // Draw value text on
       const uint8_t fs = DWINUI::fontWidth(meshfont);
       if (viewer_print_value) {
-        xy_int8_t offset;
-        offset.y = cell_height_px / 2 - fs;
+        xy_uint_t offset { 0, cell_height_px / 2 - fs };
         if (isnan(bedlevel.z_values[x][y])) {  // undefined
           dwinDrawString(false, meshfont, COLOR_WHITE, COLOR_BG_BLUE, start_x_px + cell_width_px / 2 - 5, start_y_px + offset.y, F("X"));
         }

--- a/Marlin/src/lcd/e3v2/proui/bedlevel_tools.cpp
+++ b/Marlin/src/lcd/e3v2/proui/bedlevel_tools.cpp
@@ -247,7 +247,8 @@ bool BedLevelTools::meshValidate() {
       // Draw value text on
       const uint8_t fs = DWINUI::fontWidth(meshfont);
       if (viewer_print_value) {
-        xy_int_t offset { 0, cell_height_px / 2 - fs };
+        xy_int8_t offset;
+        offset.y = cell_height_px / 2 - fs;
         if (isnan(bedlevel.z_values[x][y])) {  // undefined
           dwinDrawString(false, meshfont, COLOR_WHITE, COLOR_BG_BLUE, start_x_px + cell_width_px / 2 - 5, start_y_px + offset.y, F("X"));
         }

